iree/hal/drivers/CMakeLists.txt

```diff
@@ -39,5 +44,7 @@ iree_cc_library(
     iree::base::api
     iree::base::tracing
     ${IREE_HAL_DRIVER_MODULES}
+    $<$<PLATFORM_ID:Darwin>:${Metal}>
+    $<$<PLATFORM_ID:Darwin>:${Foundation}>
```

Comment:
Same here - can this go in https://github.com/google/iree/blob/main/iree/hal/metal/CMakeLists.txt and/or https://github.com/google/iree/blob/main/iree/hal/metal/registration/CMakeLists.txt instead?